FRUT caused Visual Studio to generate LNK1104 linker error when adding extra linker flags to Projucer‘s Visual Studio exporter

I met a problem here.
When I add a extra linker flag "\ignore:4049" to Projucer‘s Visual Studio exporter, and build the Visual Studio solution generated by FRUT and CMake, the VS generates a fatal link error: "LINK : fatal error LNK1104: cannot open file '\ignore:4049.obj'".
Obviously, there is something wrong with FRUT parsing the extra linker flag incorrectly.
So can you fix it?
Thanks!

--- a/cmake/Reprojucer.cmake
+++ b/cmake/Reprojucer.cmake
@@ -4986,7 +4986,11 @@ function(_FRUT_set_compiler_and_linker_settings target)
     endforeach()
   endforeach()
 
-  target_link_libraries(${target} PRIVATE ${JUCER_EXTRA_LINKER_FLAGS})
+  foreach(flag IN LISTS JUCER_EXTRA_LINKER_FLAGS)
+    set_property(TARGET ${target} APPEND_STRING PROPERTY LINK_FLAGS " ${flag}")
+    set_property(TARGET ${target} APPEND_STRING PROPERTY STATIC_LIBRARY_FLAGS " ${flag}")
+  endforeach()
+
   target_link_libraries(${target} PRIVATE ${JUCER_EXTERNAL_LIBRARIES_TO_LINK})
 
 endfunction()

--- a/cmake/Reprojucer.cmake
+++ b/cmake/Reprojucer.cmake
@@ -5056,8 +5056,9 @@ function(_FRUT_set_compiler_and_linker_settings_APPLE target)
           target_compile_options(${target} PRIVATE
             $<$<CONFIG:${config}>:-stdlib=${JUCER_CXX_LIBRARY_${config}}>
           )
+          string(TOUPPER "${config}" upper_config)
           set_property(TARGET ${target} APPEND_STRING PROPERTY
-            LINK_FLAGS " $<$<CONFIG:${config}>:-stdlib=${JUCER_CXX_LIBRARY_${config}}>"
+            LINK_FLAGS_${upper_config} " -stdlib=${JUCER_CXX_LIBRARY_${config}}"
           )
         endif()
       endforeach()

